api/app/Models/PoolCandidateSearchRequest.php

```diff
@@ -67,7 +68,8 @@ protected static function boot()
 
         parent::boot();
         static::creating(function ($searchRequest) use ($user) {
-            $searchRequest->user_id = $user?->id;
+            // unless the user_id was already specified use the currently authenticated user
+            $searchRequest->user_id = $searchRequest->user_id ?? $user?->id;
```

Comment:
This change was needed to be able to set the user_id in a factory.

Reply:
Could you mention that in the comment, actually?